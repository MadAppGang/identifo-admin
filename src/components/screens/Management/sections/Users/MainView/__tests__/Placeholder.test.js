import UsersPlaceholder from '../Placeholder';

describe('<UsersPlaceholder />', () => {
  let props;

  beforeEach(() => {
    props = {
      onCreateNewUserClick: jest.fn(),
    };
  });

  test('renders as expected', () => {
    expect(shallow(<UsersPlaceholder {...props} />)).toMatchSnapshot();
  });

  test('calls props onClick function on button click', () => {
    shallow(<UsersPlaceholder {...props} />).find('Button').simulate('click');
    expect(props.onCreateNewUserClick).toBeCalled();
  });
});
