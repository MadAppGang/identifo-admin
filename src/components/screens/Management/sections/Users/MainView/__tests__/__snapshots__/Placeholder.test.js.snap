// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersPlaceholder /> renders as expected 1`] = `
<div
  className="iap-section-placeholder"
>
  <h2
    className="iap-section-placeholder__title"
  >
    Users
  </h2>
  <UsersIcon
    className="iap-section-placeholder__icon"
  />
  <p
    className="iap-section-placeholder__msg"
  >
    No users have been added to your applications.
  </p>
  <Button
    Icon={[Function]}
    disabled={false}
    error={false}
    iconClassName=""
    onClick={[MockFunction]}
    stretch={false}
    transparent={false}
    type="button"
  >
    Create your first user
  </Button>
</div>
`;
